### ISSUE TITLE
Resize pdf content

### ISSUE THREAD

**Issue by ryanfigueiredo** [Hopding/pdf-lib#319]
I wish pdf-lib could resize the content. Same as with pictures in 'paint' for example. It would be possible?

**Comment by syonfox** [comment]
isn't that what scale does If i understand correctly.. scale is just a useful function to compute the dimensions of your scaled image. x and y are the offset from the bottom left in pt and with and height are the dimension the image should be drawn.. it will stretch the image to meat these.. note that the aspect ratio does not have to remain constant and you can put any numbers there. here is an snip it.  if you are asking about coping images when resizing rather then scaling them then i don't thing that's possible yet but that would be a useful feature.
```
    const pngImage = await pdfDoc.embedPng(img);
    const pngDims = pngImage.scale(0.5);
    console.log(pngImage);

    firstPage.drawImage(pngImage, {
        x: 44.201,
        y: 44.201,
        width: pngDims.width,
        height: pngDims.height ,
    });
```

**Comment by Hopding** [comment]
Hello @ryanfigueiredo!

@syonfox is correct. You do not need to preserve the aspect ratio of embedded images. You can pass in whatever width and height you want.

If you're trying to do something else, you'll need to ask something more specific. Examples and screenshots would help to clarify your question.

**Comment by tw1t611** [comment]
I came across this issue while looking for a way to compress the overall pdf file size.
Turns out, the main reason for file size increase were embedded images.

I am running my backend on `gcp`, so `ghostscript` was not an option.
Therefor I resized the images using `sharp`
```bash
yarn add sharp
```

```js
const sharp = require("sharp");

const resizedFileBuffer = await sharp(fileBuffer).resize(540, 340).toBuffer();
const embeddedJpg = await pdfDoc.embedJpg(resizedFileBuffer)
```

I was able to reduce filesize from 19.1 MB to 3.6 MB. The pdf template I used already had a size of 3.3 MB.

Hope that helps somebody.
